src/documentation/sdks/python-pipeline-dependencies.md

```diff
@@ -0,0 +1,123 @@
+
+3. Run your pipeline with the following command-line option:
+
+        --requirements_file requirements.txt
+
+    The runner will use the `requirements.txt` file to install your additional dependencies onto the remote workers.
+
+**Important:** Remote workers will install all packages listed in the `requirements.txt` file. Because of this, it's very important that you delete non-PyPI packages from the `requirements.txt` file, as stated in step 2. If you don't remove non-PyPI packages, the remote workers attempt to install packages from sources that are unknown to them, and an error occurs.
+
+
+## <a name="localnonpypi"></a>Local or non-PyPI Dependencies
+
+If your pipeline uses packages that are not available publicly (e.g. packages that you've downloaded from a GitHub repo), make these packages available remotely by performing the following steps:
+
+1. Identify which packages are installed on your machine and are not public. Run the following command:
+
+        pip freeze
+
+    This command lists all packages that are installed on your machine, regardless of where they were installed from.
+
+2. Run your pipeline with the following command-line option:
+
+        --extra_package /path/to/package/package-name
+
+
+## <a name="multfiles"></a>Multiple File Dependencies
+
+Often, your pipeline code spans multiple files. To run your project remotely, you must group these files as a Python package. When remote workers spin up, they look for this kind of package in the staging location and install it. To group your files as a Python package and make it available remotely, perform the following steps:
+
+1. Create a [setup.py](https://pythonhosted.org/an_example_pypi_project/setuptools.html) file for your project. The following is a very basic `setup.py` file.
+
+        setuptools.setup(
+           name='PACKAGE-NAME'
+           version='PACKAGE-VERSION',
+           install_requires=[],
+           packages=setuptools.find_packages(),
+        )
+
+2. Structure your project so that the root directory contains the `setup.py` file, the main workflow file, and a directory with the rest of the files.
+
+        root_dir/
+          setup.py
+          main.py
+          other_files_dir/
+
+    See [Juliaset](https://github.com/apache/beam/tree/master/sdks/python/apache_beam/examples/complete/juliaset) for an example that follows this required project structure.
+
+3. Run your pipeline with the following command-line option:
+
+        --setup_file /path/to/setup.py
+
+**Note:** If you [created a requirements.txt file](#pypi) and your project spans multiple files, you can get rid of the `requirements.txt` file and instead, add all packages contained in `requirements.txt` to the `install_requires` field of the setup call (in step 1).
+
+
+## <a name="nonpython"></a>Non-Python Dependencies
```

Comment:
I think this section and the next one (`PyPI Dependencies with Non-Python Dependencies`) can be merged.

We can remove this section and change the title of the next one to `Non-Python Dependencies and PyPI Dependencies with Non-Python Dependencies`. (Also maybe some adjustment to the first sentence to cover both cases.)